### PR TITLE
Evapor additions melt pool

### PR DESCRIPTION
**Pull request by mschreter**
This PR includes:

- fix melt pool with evaporation (no recoil pressure force is added in case of evaporation) 
- introduction of an additional case for computing the density in the interfacial region consistent with the mass balance
- update to C++20 standard

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ MESSAGE("_______________________________________________________________________
 
 # this is the standard deal.II search mechanism, including check for Trilinos and p4est
 cmake_minimum_required(VERSION 3.3)  
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 20)
```
Comment (peterrum) [inline CMakeLists.txt:14]: What is the reason to go to C++20?

File: include/meltpooldg/flow/two_phase_flow_problem.hpp [hunk 2]
```diff
@@ -518,24 +518,30 @@ namespace MeltPoolDG::Flow
 
               for (unsigned int q = 0; q < ls_values.n_q_points; ++q)
                 {
-                  const auto indicator = parameters.flow.variable_properties_over_interface ?
-                                           ls_values.get_value(q) :
-                                           UtilityFunctions::heaviside(ls_values.get_value(q), 0.5);
+                  const auto indicator =
+                    parameters.flow.variable_properties_over_interface == "true" ||
+                        parameters.flow.variable_properties_over_interface ==
+                          "consistent_with_evaporation" ?
```
Comment (peterrum) [inline include/meltpooldg/flow/two_phase_flow_problem.hpp:524]: Let's do this string compare just once (outside of the loops).

File: include/meltpooldg/interface/parameters.hpp [hunk 3]
```diff
@@ -674,7 +675,7 @@ namespace MeltPoolDG
         prm.add_parameter(
           "laser variable properties over interface",
           laser.variable_properties_over_interface,
-          "Set this parameter to true to interpolate the thermal properties over the interface smoothly.");
+          "Set this parameter to true interpolate the thermal properties over the interface smoothly.");
```
Comment (peterrum) [inline include/meltpooldg/interface/parameters.hpp:678]: Have you removed the wrong "to"?

**Review by peterrum — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (mschreter) [reply CMakeLists.txt:14]: I have also activated C++20 standard in deal.II. What are your concerns?

**Pull request merged**